fingerprint2.js

```diff
@@ -21,1431 +21,1408 @@
+      var names = [
+        'AcroPDF.PDF', // Adobe PDF reader 7+
+        'Adodb.Stream',
+        'AgControl.AgControl', // Silverlight
+        'DevalVRXCtrl.DevalVRXCtrl.1',
+        'MacromediaFlashPaper.MacromediaFlashPaper',
+        'Msxml2.DOMDocument',
+        'Msxml2.XMLHTTP',
+        'PDF.PdfCtrl', // Adobe PDF reader 6 and earlier, brrr
+        'QuickTime.QuickTime', // QuickTime
+        'QuickTimeCheckObject.QuickTimeCheck.1',
+        'RealPlayer',
+        'RealPlayer.RealPlayer(tm) ActiveX Control (32-bit)',
+        'RealVideo.RealVideo(tm) ActiveX Control (32-bit)',
+        'Scripting.Dictionary',
+        'SWCtl.SWCtl', // ShockWave player
+        'Shell.UIHelper',
+        'ShockwaveFlash.ShockwaveFlash', // flash plugin
+        'Skype.Detection',
+        'TDCCtl.TDCCtl',
+        'WMPlayer.OCX', // Windows media player
+        'rmocx.RealPlayer G2 Control',
+        'rmocx.RealPlayer G2 Control.1'
+      ]
         // starting to detect plugins in IE
-        result = this.map(names, function (name) {
-          try {
+      result = map(names, function (name) {
+        try {
             // eslint-disable-next-line no-new
-            new window.ActiveXObject(name)
-            return name
-          } catch (e) {
-            return null
-          }
-        })
-      }
-      if (navigator.plugins) {
-        result = result.concat(this.getRegularPlugins())
-      }
-      return result
-    },
-    pluginsShouldBeSorted: function () {
-      var should = false
-      for (var i = 0, l = this.options.sortPluginsFor.length; i < l; i++) {
-        var re = this.options.sortPluginsFor[i]
-        if (navigator.userAgent.match(re)) {
-          should = true
-          break
+          new window.ActiveXObject(name)
+          return name
+        } catch (e) {
+          return null
         }
+      })
+    }
+    if (navigator.plugins) {
+      result = result.concat(getRegularPlugins(options))
+    }
+    return result
+  }
+  var pluginsShouldBeSorted = function (options) {
+    var should = false
+    for (var i = 0, l = options.sortPluginsFor.length; i < l; i++) {
+      var re = options.sortPluginsFor[i]
+      if (navigator.userAgent.match(re)) {
+        should = true
+        break
       }
-      return should
-    },
-    touchSupportKey: function (keys) {
-      if (!this.options.excludeTouchSupport) {
-        keys.addPreprocessedComponent({key: 'touch_support', value: this.getTouchSupport()})
-      }
-      return keys
-    },
-    hardwareConcurrencyKey: function (keys) {
-      if (!this.options.excludeHardwareConcurrency) {
-        keys.addPreprocessedComponent({key: 'hardware_concurrency', value: this.getHardwareConcurrency()})
-      }
-      return keys
-    },
-    hasSessionStorage: function () {
-      try {
-        return !!window.sessionStorage
-      } catch (e) {
-        return true // SecurityError when referencing it means it exists
-      }
-    },
-    // h
```

Comment:
I don't understand exactly, what assignment needs to moved ?

Reply:
Sorry. The `touchEvent = false` assignment. The code tests for `touchEvent` availability; in case it's available the assignment is near the test, otherwise it's at the start of the function, would be easier to understand if both are near the test.